### PR TITLE
fix: add DISPLAY_NAME_LENGTH to be configurable

### PR DESCRIPTION
**Pull request by adong**
## Context

Successor of https://github.com/screwdriver-cd/ui/pull/588

## Objective
- allow users to customize display name length for their pipelines
- integrated with [ember-local-storage](https://github.com/funkensturm/ember-local-storage) to buff up localStorage usage.

See screenshot:
image

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by adong** [comment]
rebase with latest master

File: app/components/pipeline-options/template.hbs [hunk 1]
```diff
@@ -153,6 +153,39 @@
...
+            <div class="col-xs-10">
+              <p>All settings here are stored locally in your browser.</p>
+            </div>
+          </div>
+        </li>
+        <li>
+          <div class="row">
+            <div class="col-xs-10">
+              <h4>Display Name Length</h4>
+              <p>Setup your own preferred length to display on the workflow graph</p>
```
[the remark this answers is not in the document]
Comment (adong) [reply app/components/pipeline-options/template.hbs]: updated

File: app/components/workflow-graph-d3/component.js [hunk 2]
```diff
@@ -135,11 +137,23 @@ export default Component.extend({
       }
     });
   },
-  draw(data) {
-    const MAX_DISPLAY_NAME = 20;
+  async draw(data) {
+    let DISPLAY_JOBNAME_LENGTH = ENV.APP.MINIMUM_JOBNAME_LENGTH;
```
Comment (pritamstyz4ever) [inline app/components/workflow-graph-d3/component.js:141]: why is it upper case?

**Review by pritamstyz4ever — DISMISSED** [review]
minor comment

File: app/components/workflow-graph-d3/component.js (hunk 2 above)
Comment (adong) [reply app/components/workflow-graph-d3/component.js]: Yeah. It's not necessary to be uppercase. I guess the original variable is a `const`, and refactored code is `let`. 

Updated

**Review by jithine — APPROVED** [review]

**Pull request merged**